### DISCUSSION TITLE
Adding a screen (table block) for a particular postgreSQL table does not work as expected

### DISCUSSION THREAD

**Discussion by kpturner** [Budibase/budibase#11017] [Help] [answered]
For all my tables (except one) if I create a screen and take the `Quickly create a screen from your data` option, I ended up with a functional table that lists all the rows which I can then start to augment to my needs. For example:
![image](https://github.com/Budibase/budibase/assets/308646/80a1b574-c4cb-4b58-a56b-8b2e9ea1beda)
  

For one particular table, however, it does not do that. Instead of a table I get this:
![image](https://github.com/Budibase/budibase/assets/308646/66cef063-e378-4980-8f54-1ef3fa98e274)

I cannot see anything particularly unusual about this table.  It just has a few fields and a primary key constraint (`iso` column) so what would cause this?

**Comment by kpturner** [comment]
It seems that the default column width (`Auto`) sets the width of the column to be the full width or the container for some reason.  Why?

**Comment by kpturner** [comment]
Strangely, if I "Configure Columns" and add them all, then set the all to a width of 200px, it does at least render the table nicely.  If I then reset them one by one back to "Auto" it remains fine.  

Is this a bug?

**Reply by ConorWebb96** [reply]
Hey @kpturner,

Can I ask which browser you are using? 

This would definitely be a bug, are you able to pull together some replication steps? I've tried replicate on my end and wasn't able to. (I've had a tried on both Firefox and Chrome.)

The styling within inspect looks like this. Could you verify this is also what you are seeing when everything is set to auto. When you set the width you will see that this is were it gets amended.
<img width="213" alt="Screenshot 2023-06-28 at 11 22 21" src="https://github.com/Budibase/budibase/assets/126772285/03bdf2f2-9df5-4c97-b891-3eda126df209">

If you can replicate I'd recommend filling out a bug template within in issues, we will verify and get it swiftly fixed.

Thank you and I hope this information helps.

**Reply by kpturner** [reply]
What element are you inspecting?

**Reply by ConorWebb96** [reply]
Apologies, its the div with the class; spectrum-Table 
<img width="918" alt="Screenshot 2023-06-28 at 11 45 24" src="https://github.com/Budibase/budibase/assets/126772285/6ff421ca-901a-49c2-97a4-7d6c37b2f980">

**Reply by kpturner** [reply]
Well if I add a new table block component for this table without doing anything to configure the columns, they are again a mess and I see (invalid CSS). No idea where 200 comes from. 
![image](https://github.com/Budibase/budibase/assets/308646/5c8bb16a-968b-4c57-8e1f-3b5015a32043)

With the component I modified to make it work again with `Auto` I see 
![image](https://github.com/Budibase/budibase/assets/308646/e86fcb7d-dde4-424b-993e-20b7de13d499)

**Reply by kpturner** [reply]
If I manually change the widths from `Auto` to `minmax(auto, 1fr)` then it renders ok and I am back to
![image](https://github.com/Budibase/budibase/assets/308646/73f84d33-d7cb-41a6-a125-626dda8e8237)

**Reply by kpturner** [reply]
On a different table if I add a table block component I also get 
![image](https://github.com/Budibase/budibase/assets/308646/266bbf5d-84e1-4ad5-a90d-c77130438dd7)
out of the box.   So something in the `country` table is corrupted with regard to column widths.   If I set it to `Auto` I do not end up with `minmax(auto, 1fr)`, I end up with `Auto`.  If I don't touch anything I end up with `200`

**Reply by ConorWebb96** [reply]
Hey @kpturner,

Could I get the schema for the table that is being affected. Its very strange that its only affecting one of your tables and not the others. I'm assuming you are using the Postgres data source?

Does the issue still persist when you eject the table block component?

An app export would be great also so we can have your exact settings! Feel free to email it to support@budibase.com if you don't want  to attach it here.

Thanks for the information!

**Comment by kpturner** [comment, marked as answer]
Bug report:  https://github.com/Budibase/budibase/issues/11051

**Reply by ConorWebb96** [reply]
Could you also provide a sample app export and schema for the table in the bug report, so we have an example to work from to fix!

Thank you!

**Reply by kpturner** [reply]
If I export the app, will it not contain also the password for the datasource?  I cannot provide that.  It begs the question, can you see that by looking at the hosted app?

**Reply by ConorWebb96** [reply]
You are right it will contain the passwords within the app export, the only way around this would be if you were business tier and used env variables. 

As you are on the free tier. Could you provide the table's schema thats causing this to happen. So we can try and replicate, feel free to attach this onto the bug report you created! As it will be very helpful for us, it doesn't have to include any data.

Thanks again for raising this!

**Comment by andz-bb** [comment]
Hi @kpturner, thanks for sending us your db schema. I spent some time investigating your issue but unfortunately I have not been able to replicate it on my end. 

I imported your schema into my local Postgres server and created a new app with that as the datasource. When I generated a screen for the empty country table, the table rendered properly and was formatted correctly.

I then configured the country-currency relationship in Budibase and added a test row to each table to make sure the data itself wasn't causing problems. When I deleted and regenerated the screen it still looked OK.

![country screen autogenerated](https://github.com/Budibase/budibase/assets/110921612/43213065-4b20-4e3e-8ada-f3333acffbca)

Does the issue still occur if you re-fetch the tables in your Postgres datasource settings and then regenerate the screen/table? Are you able to replicate it using the same db in a brand new app? Thanks

**Reply by kpturner** [reply]
I think I mentioned this elsewhere but re-importing the tables makes no difference.  It seems that once BB has hold of the table and has populated its own internal DB with info about the table, it has you on toast.  There is no facility to completely delete a table from BB either.  Once it has been imported, it is impossible to get rid of it there is a separate discussion about that).

I will try another app later

**Reply by kpturner** [reply]
Update:  There is no problem on a new app.

**Reply by andz-bb** [reply]
thanks for the quick response @kpturner 

Just to double check, does it still hang on to that info when you delete your screens and the whole Postgres datasource and then re-add them?

**Reply by kpturner** [reply]
Deleting the datasource completely is rather inconvenient in that I lose all my changes and I am not sure what happens to the screens using that datasource.  Nevertheless I deleted them all and the datasource and started again.  That worked.

**Reply by kpturner** [reply]
I have just had this happen again with another table.  This time the table was all fine (screens deigned and save - all OK).
I then wanted to check the type of one particular field. So I went to visit the grid in the `Data` section (as opposed to `Design`) edited one of the columns to ensure it was `Number`, clicked save and the returned to the design.  This was the result for the table page:

![image](https://github.com/Budibase/budibase/assets/308646/205eb8b9-cc35-4371-a894-6372fbeebabc)

Deleting and reimporting the schema is not an option now.
